[CIAPP] Extend Jenkins documentation for CI Visibility using plugin UI (#10973)
DataDog/documentation: content/en/continuous_integration/setup_pipelines/jenkins.md

## Install the Datadog Jenkins plugin

Install and enable the [Datadog Jenkins plugin][2] v3.0.0 or newer:

1. In your Jenkins instance web interface, go to **Manage Jenkins > Manage Plugins**.
2. In the [Update Center][3] on the **Available** tab, search for `Datadog Plugin`.
3. Select the checkbox next to the plugin, and install using one of the two install buttons at the bottom of the screen.

> 4. To verify the plugin is installed, search for `Datadog Plugin` on the **Installed** tab.

Reviewer: ```suggestion
4. To verify that the plugin is installed, search for `Datadog Plugin` on the **Installed** tab.
```

## Enabling CI Visibility on the plugin

1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.
2. Go to the `Datadog Plugin` section, scrolling down the configuration screen.
3. Select the `Datadog Agent` mode.
4. Configure the `Datadog Agent` host.

> 5. Configure the `Traces Collection` port. (By default `8126` in the Datadog Agent)

Reviewer 2: ```suggestion
5. Configure the `Traces Collection` port. (The default port is `8126` in the Datadog Agent.)
```

6. Click on `Enable CI Vibisility` checkbox to activate it.
7. Configure your CI Instance name.
8. Check the connectivity with the Datadog Agent.
9. Save your configuration.
10. To verify that CI Visibility is enabled, go to `Jenkins Log` and search for:

{{< code-block lang="text" >}}
Re/Initialize Datadog-Plugin Tracer: hostname = <HOST>, traceCollectionPort = <TRACE_PORT>
DATADOG TRACER CONFIGURATION {...}
{{< /code-block >}}

> **Note**: Enabling CI Visibility using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.

Reviewer: ```suggestion
**Note**: Enabling CI Visibility using the Jenkins plugin is not compatible with running the Java APM tracer as a Java agent when launching Jenkins.
```

## Enable job log collection in the plugin

You can correlate your pipeline logs with the traces generated by CI Visibility.

> 1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.

Reviewer: ```suggestion
1. In the web interface of your Jenkins instance, go to **Manage Jenkins > Configure System**.
```

You can configure the Jenkins Plugin to send custom tags in all pipeline traces:

> 1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.

Reviewer: same as above

> 1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.

Reviewer 2: ```suggestion
1. In the web interface of your Jenkins instance, go to **Manage Jenkins > Configure System**.
```

2. Go to the `Datadog Plugin` section, scrolling down the configuration screen.
3. Click on the `Advanced` button.
4. Configure the `Global Tags`.
5. Configure the `Global Job Tags`.
6. Save your configuration.

**Global tags** (`globalTags`): A comma-separated list of tags to apply to all metrics, traces, events, and service checks. Tags can include environment variables that are defined in the Jenkins controller instance: `global_tag:global_value,global_tag2:${SOME_ENVVAR},${OTHER_ENVVAR}:global_tagvalue`

> **Global job tags** (`globalJobTags`): A comma-separated list of regex to match a job and a list of tags to apply to that job. Tags can include environment variables that are defined in the Jenkins controller instance. Tags can reference match groups in the regex using the `$` symbol: `(.*?)_job_(*?)_release:someValue`

Reviewer 2: ```suggestion
**Global job tags** (`globalJobTags`): A comma-separated list of regexes to match a job and a list of tags to apply to that job. Tags can include environment variables that are defined in the Jenkins controller instance. Tags can reference match groups in the regex using the `$` symbol: `(.*?)_job_(*?)_release:someValue`
```

## Visualize pipeline data in Datadog

Once the integration is successfully configured, both [Pipelines][6] and [Pipeline Executions][7] pages will start populating with data after pipelines finish.

**Note**: The Pipelines page shows data for only the default branch of each repository.

## Troubleshooting

> ### Configure DEBUG log level for Datadog Plugin logs

Reviewer: ```suggestion
### Enable DEBUG log level for the Datadog Plugin
```

> If you have any issue with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you will be able to check the exception stacktrace if some issue happens.

Reviewer: ```suggestion
If you have any issue with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you will be able to check the exception stacktrace if there are any issues.
```
or any issue happens, but pretty sure some is wrong here

> If you have any issue with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you will be able to check the exception stacktrace if some issue happens.

Reviewer 2: ```suggestion
If you have any issues with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you are able to check the exception stacktrace if an issue arises.
```

1. In your Jenkins instance web interface, go to **Manage Jenkins > System log**.
2. Click on `Add new log recorder` button.
3. Type the log recorder name. E.g: **Datadog Plugin Logs**.
4. Add the following loggers to the list:
- Logger: `org.datadog.jenkins.plugins.datadog.clients` -> Log Level `ALL`
- Logger: `org.datadog.jenkins.plugins.datadog.traces` -> Log Level `ALL`
- Logger: `org.datadog.jenkins.plugins.datadog.logs` -> Log Level `ALL`
- Logger: `org.datadog.jenkins.plugins.datadog.model` -> Log Level `ALL`
- Logger: `org.datadog.jenkins.plugins.datadog.listeners` -> Log Level `ALL`
5. Save the configuration.

You may also want to split the loggers into different log recorders.

> Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing to the desired log recorder via **Manage Jenkins > System log**.

Reviewer: ```suggestion
Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing the desired log recorder via **Manage Jenkins > System log**.
```

> Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing to the desired log recorder via **Manage Jenkins > System log**.

Reviewer 2: ```suggestion
Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing the desired log recorder through **Manage Jenkins > System log**.
```

> If you triggered some Jenkins pipelines, you can search for `Send pipeline traces` message in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.

Reviewer: ```suggestion
If you trigger a Jenkins pipelines, you can search for the message `Send pipeline traces` in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.
```

> If you triggered some Jenkins pipelines, you can search for `Send pipeline traces` message in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.

Reviewer 2: ```suggestion
If you trigger a Jenkins pipeline, you can search for the message `Send pipeline traces` in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.
```

{{< code-block lang="text" >}}
Send pipeline traces.
...
Send pipeline traces.
...
{{< /code-block >}}

### The Datadog Plugin section does not appear in the Jenkins configuration

> If the Datadog Plugin section does not appear in Jenkins configuration section, you need to make sure that the plugin is enabled. To do so:

Reviewer 2: ```suggestion
If the Datadog Plugin section does not appear in Jenkins configuration section, make sure that the plugin is enabled. To do so:
```

1. In your Jenkins instance web interface, go to **Manage Jenkins > Manage Plugins**.
2. Search for `Datadog Plugin` in the **Installed** tab.
3. Check that the `Enabled` checkbox is marked.

> 4. If you enable the plugin here, restart your Jenkins instance using `/safeRestart` URL path.

Reviewer 2: ```suggestion
4. If you enable the plugin here, restart your Jenkins instance using the `/safeRestart` URL path.
```

### The Plugin Tracer failed to initialized due to APM Java Tracer is used to instrument Jenkins.

> If this error message appears in the **Jenkins Log**, you need to make sure that you are not using the APM Java Tracer to instrument your Jenkins instance.

Reviewer 2: ```suggestion
If this error message appears in the **Jenkins Log**, make sure that you are not using the APM Java Tracer to instrument your Jenkins instance.
```

{{< code-block lang="text" >}}
Failed to reinitialize Datadog-Plugin Tracer, Cannot enable traces collection via plugin if the Datadog Java Tracer is being used as javaagent in the Jenkins startup command. This error will not affect your pipelines executions.
{{< /code-block >}}

> 1. Access to your Jenkins controller instance machine via console.

Reviewer 2: ```suggestion
1. Access your Jenkins controller instance machine using the console.
```

2. Check the Java startup line for the Jenkins process.
{{< code-block lang="shell" >}}
$> ps -ef | grep jenkins
{{< /code-block >}}
3. Ensure you are not using the APM Java Tracer as `-javaagent`.

> **Note**: Enabling CI Visibility using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.

Reviewer: not compatible

> **Note**: Enabling CI Visibility using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.

Reviewer 2: ```suggestion
**Note**: Enabling CI Visibility using the Jenkins plugin is not compatible with running the Java APM tracer as a Java agent when launching Jenkins.
```
